go/libraries/doltcore/doltdb/commit_itr.go

```diff
@@ -0,0 +1,135 @@
+package doltdb
+
+import (
+	"context"
+	"errors"
+	"github.com/liquidata-inc/dolt/go/store/hash"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+// CommitItr is an interface for iterating over a set of unique commits
+type CommitItr interface {
+	// Next returns the hash of the next commit, and a pointer to that commit.  Implementations of Next must handle
+	// making sure the list of commits returned are unique.  When complete Next will return hash.Hash{}, nil, nil
```

Comment:
Returning io.EOF is more idiomatic for iterators in go, right?

Reply:
done